### PR TITLE
✨Data model changes

### PR DESCRIPTION
**Pull request by parimalak**
- [x] model changes
- [x] update erd
- [x] update migrations
Addresses #406 #391 #400

### CODE REVIEW

File: dataservice/api/family_relationship/models.py [hunk 1]
```diff
@@ -57,6 +58,8 @@ class FamilyRelationship(db.Model, Base):
                                                       nullable=False)
 
     participant2_to_participant1_relation = db.Column(db.Text())
+    source_text_notes = db.Column(db.Text(),
```
Comment (dankolbman) [inline dataservice/api/family_relationship/models.py:62]: What kind of notes are these exactly?

File: dataservice/api/family/schemas.py [hunk 2]
```diff
@@ -1,9 +1,16 @@
+from marshmallow_sqlalchemy import field_for
 from dataservice.api.family.models import Family
 from dataservice.api.common.schemas import BaseSchema
 from dataservice.extensions import ma
+from dataservice.api.common.validation import (enum_validation_generator)
+FAMILY_TYPE_ENUM = {"Proband Only", "Duo", "Duo+",
+                    "Trio", "Trio +", "Other", }
```
Comment (znatty22) [inline dataservice/api/family/schemas.py]: Should probably be `Trio+` right? Looks like theres a space right now between Trio and +

File: dataservice/api/family_relationship/models.py (hunk 1 above)
Comment (znatty22) [reply dataservice/api/family_relationship/models.py:62]: I haven't seen it very often, but I think they're supposed to be anything the investigator included in their pedigree file describing the relationship.

Comment (parimalak) [reply dataservice/api/family_relationship/models.py:62]: ya, for example, Engle data has special notes for some relations.

**Review by znatty22 — APPROVED** [review]
🥇

**Review by dankolbman — APPROVED** [review]
lgtm 🎉

**Pull request merged**